pkg/tfsandbox/destroy.go

```diff
@@ -2,17 +2,95 @@ package tfsandbox
 
 import (
 	"context"
-	"fmt"
+
+	tfjson "github.com/hashicorp/terraform-json"
 )
 
 // Destroy runs the terraform destroy command
-func (t *Tofu) Destroy(ctx context.Context, log Logger) error {
+// This handles a couple of special cases around destroy.
+// The returned State and Error can both be not nil. This would be the case
+// if the destroy failed and some resources were destroyed and some were not.
+//
+// There are two special cases that callers should be aware of:
+//  1. If the destroy was successful, we return a non-nil state with no resources
+//  2. If the destroy failed, we either return the state with the resources that failed to destroy
+//     or a nil state if something went really wrong and there is not state left
+func (t *Tofu) Destroy(ctx context.Context, log Logger) (*State, error) {
+	rawState, success, destroyErr := t.destroy(ctx, log)
+	if success {
+		s, err := newDestroyState(rawState)
+		if err != nil {
+			return nil, err
+		}
+		return s, nil
+	}
+	if rawState != nil {
+		s, err := newDestroyState(rawState)
+		if err != nil {
+			return nil, err
+		}
+		return s, destroyErr
+	}
+	return nil, destroyErr
+}
+
+// Destroy runs the terraform destroy command
+// This handles a couple of special cases around destroy.
+//  1. If the destroy was successful, we return a nil state, success = true, nil error
```

Comment:
This seems to assume that successful `tofu destroy` always produces a state equivalent to a nil state, yes? That may be actually true. If we were to not assume it we could call `Show` anyway and check if the state is empty. Probably OK to assume it though, given what we know.